pkg/controller/jobs/pod/pod_controller.go

```diff
@@ -736,10 +736,19 @@ func (p *Pod) notRunnableNorSucceededPods() []corev1.Pod {
 	return utilslices.Pick(p.list.Items, func(p *corev1.Pod) bool { return !isPodRunnableOrSucceeded(p) })
 }
 
+func isUnschedulable(p *corev1.Pod) bool {
+	for _, c := range p.Status.Conditions {
```

Comment:
Generally the one problem I see with this is that we don't have any indication that an attempt to assign a node was made, however since someone or something has deleted the pod I think we can live with it.

Reply:
> Or even just the phase Pending.

I was thinking about it too, but this scenario is possible:
1. kubelet has already started the containers, and is about to transition `Pending` -> `Running`
2. The user deletes the pod so we have `deletionTimestamp != nil` and `Pending`
3. The patch by kubelet succeeds and the pod is `Running`
4. The containers complete successfully within graceful pod period  and kubelet transitions the pod to `Succeeded`

This is **super** unlikely, but possible. I'm ok either way, but maybe we stay on the safe side for now, just 
patching for the reported issue.